cloud/services/publicips/service.go

```diff
@@ -26,7 +26,7 @@ import (
 type PublicIPScope interface {
 	logr.Logger
 	azure.ClusterDescriber
-	PublicIPSpecs() []azure.PublicIPSpec
+	PublicIPSpecs() ([]azure.PublicIPSpec, error)
```

Comment:
I think you might be able to get rid of `error` on this signature.

Reply:
it's from the hash in the name generation: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/652/files#diff-7c0af1847b5ee759aadb398ac16c4d36R124

are you thinking i'd move that elsewhere?

Reply:
I get it 👍